lib/lz4frame.c

```diff
@@ -134,12 +134,14 @@ static int g_debuglog_enable = 1;
   typedef uint32_t U32;
   typedef  int32_t S32;
   typedef uint64_t U64;
+  typedef uintptr_t uptrval;
```

Comment:
Unfortunately `uintptr_t` is not guaranteed to be present in `<stdint.h>`.
This could lead to portability issues for platforms that don't support it.

Reply:
You're right.  In the standard, it's optional type.  But I've copied this definition [from `lz4.c`](https://github.com/lz4/lz4/blob/5b86e4e5ed4429a8c0f58dadc3ebaf65ba029d18/lib/lz4.c#L274-L292) for consistency.  And `uptrval` has been used in lz4.c for [long time](https://github.com/lz4/lz4/commit/1abecbc33c8ec5b84d2623dcbe73136aeb99db37#diff-7055e9cf14c488aea9837aaf9f528b58ee3c22988d7d0d81d172ec62d94a88a7R141-R152).
Therefore, if `<stdint.h>` doesn't have `uintptr_t`, they can't compile entire lz4 code at all.  Also if I change this line, we must change counterpart in lz4.c.  

But I don't say this is the best definition.  So can we leave this in this PR, and can we create other PR to address `uptrval` issue?

Although possible solution may be

```C
/* From n1256 (C99), "7.18.2 Limits of specified-width integer types"
> 1. The following object-like macros specify the minimum and
>    maximum limits of the types declared in <stdint.h>.
> ...
> - maximum value of pointer-holding unsigned integer type
>   UINTPTR_MAX

They say "the types declared in <stdint.h>".  So if uintptr_t is
not declared in <stdint.h>, they must not have a UINTPTR_MAX. */

#if defined(UINTPTR_MAX) /* UINTPTR_MAX is optional macro constant */
  typedef uintptr_t uptrval;
#else
  typedef uintmax_t uptrval; /* choose the available "best" type */
#endif

void in_some_function() {
    LZ4_STATIC_ASSERT(sizeof(uptrval) >= sizeof(void*));
}
```

Reply:
arf, I thought that I had removed that definition a long time ago already.
I might have confused with `zstd`.
Anyway, that's still a goal : we should get rid of `uintptr_t`.

Reply:
Right.  I'm also realized that `lz4frame.c` is the best candidate to start "get rid of `uintptr_t`" movement since it didn't have `uptrval` 😉